### DISCUSSION TITLE
Is it possible to switch between folders with logs in runtime?

### DISCUSSION THREAD

**Discussion by ostapkrutsko** [sevdokimov/log-viewer#117] [Q&A]
Hi sevdokimov,

I have one specialty: our log files are distributed in different folders, whose names are known only in runtime. So, how can we switch in runtime between those different folders to see only logs from one folder at a time? We are going to use the embedded in SpringBoot application version because I think it is impossible to do so in the case of a standalone application.

I must note your product is very cool, I'm trying to use it!

Thank you,
Ostap Krutsko

**Comment by sevdokimov** [comment]
You can create a spring bean implementing `com.logviewer.api.LvPathResolver` interface. This bean will be used when log-viewer transforms  a path in URL into path on the disk. 
```
public class MyLogPathResolver implements LvPathResolver {
    @Override
    public Collection<LogPath> resolvePath(String pathFromHttpParameter) {
        if (pathFromHttpParameter.equals("all-logs")) {
            List<String> allLogs = getAllLogFilesAcrossAllFolders();
            return allLogs.stream()
                    .map(filePath -> new LogPath(null, filePath))
                    .collect(Collectors.toList());
        }
        
        return null;
    }
}
```

Also you have to specify `log-viewer.accessible-files.pattern` property to make that log files visible, log-viewer hides all files by default. Example: 
```
log-viewer.accessible-files.pattern=/opt/my-app/logs/**
```

or  "config.conf"
```
log-viewer.accessible-files.pattern= [
   "/opt/my-app/logs1/*.log",
   "/opt/my-app/logs2/**.log"
]
```

If log-viewer detects log format incorrectly, you can add a spring bean implementing `com.logviewer.api.LvFormatRecognizer`
